design/fips-compliance.md

```diff
@@ -0,0 +1,213 @@
+ | Release both versions                  | <ul><li>One more Dockerfile to maintain. Any change made to the original Dockerfile should be done also to `Dockerfile.fips` which can be easily forgotten</li><li>Another version to test. Our UTs will stay the same but we will need to run integration tests on both images, in order to release them. This adds complexity to our tests and doubles the tests run-time</li></ul> |
+
+From the table above, I think that we should release only the FIPS compliant
+version. The advantages that customers will get by using the `golang` image
+as the base image is not worth the maintenance burden that releasing both
+versions will bring to the development team.
+
+### Red Hat image
+
+We will also release a FIPS compliant Red Hat image of the `conjur-authn-k8s
+-client` to our Red Hat registry. Before we do that, we should create a 
+`conjur-authn-k8s-client` FIPS project in the RH container registry.
+
+TBD - how do we create the FIPS project in the RH container registry?
+
+### Performance
+
+Same SLA of performance should be kept. As of today, we don't have any
+performance tests and an SLA (that I know of). We should decide on one and
+verify that the change to GoBoring doesn't impact the performance.
+
+We should write a performance test and run it twice - once on the `master
+` image and once on the new FIPS-compliance image - and verify that the
+ performance is not affected by the change. 
+
+## Test plan
+
+Before we dig into which tests should run, we should decide where, and how, they
+should run. Currently, the `conjur-authn-k8s-client` have UTs in its project
+and a vanilla test that runs in `kubernetes-conjur-demo`. The `kubernetes-conjur-demo`
+is triggered only when we push code to that repository. 
+
+Even while putting aside the fact that we have only a vanilla test, our current
+ test flow is still not optimal. 
+ Let's propose some options to improve our flow, and decide on the best one.
+
+### Trigger `kubernetes-conjur-demo` to run after every build of `conjur-authn-k8s-client`
+
+We can find a way to run `kubernetes-conjur-demo` using the currently built
+ authenticator-client, utilizing the `LOCAL_AUTHENTICATOR` environment variable.
+ If this is possible, we can trigger a `kubernetes-conjur-demo` build as part
+  of our build.
+
+This will give us full confidence in the green build as it also passed
+ integration tests.
+ 
+ However, the build time of `kubernetes-conjur-demo` is ~25 minutes (opposing
+  2-3 minutes of the `conjur-authn-k8s-client`) and it is not very stable
+  . This means that it will harder to merge PRs into the `conjur-authn-k8s
+  -client`.
+  
+I am not sure that this can be implemented. By setting`LOCAL_AUTHENTICATOR
+` to `true` we use the last built image (ran in the `./bin/build` step. In
+ case we have builds running in parallel it may affect the latest built image
+  so it is possible that we can't use a local authenticator in Jenkins.
```

Comment:
@hughsaunders do you think this is possible? if builds run in parallel will the image name (set in `./bin/build`) change or do the builds run in separate nodes?